agent/predictions/websocket.py

```
@@ -0,0 +1,200 @@
+
+
+class ConnectionManager:
+    """Manages WebSocket connections for market subscriptions."""
+    
+    def __init__(self):
+        # market_id -> set of websockets
+        self.active_connections: Dict[str, Set[WebSocket]] = {}
+        # all connections (for broadcast)
+        self.all_connections: Set[WebSocket] = set()
+    
+    async def connect(self, websocket: WebSocket, market_id: str = None):
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)